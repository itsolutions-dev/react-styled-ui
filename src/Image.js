import styled from 'styled-components';

const Image = styled.img`
  display: block;
  max-width: 100%;
  height: auto;
`;

export default Image;
