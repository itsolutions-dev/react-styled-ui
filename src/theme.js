export default {
  radius: 8,
  spacing: {
    margin: 8,
    padding: 8,
    iconSize: 24,
    borderSize: 1,
    shadowSize: 1,
  },
  font: {
    size: 16,
    weight: 200,
    lineHeight: 18,
    family: 'Roboto, sans-serif',
  },
  h1: {
    size: 32,
    weight: 900,
    lineHeight: 64,
  },
  h2: {
    size: 24,
    weight: 700,
    lineHeight: 32,
  },
  h3: {
    size: 18,
    weight: 500,
    lineHeight: 24,
  },
  selected: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  hover: {
    size: 0,
    weight: 10,
    lineHeight: 0,
  },
  transitions: {
    in: undefined,
    out: undefined,
  },
  palette: {
    default: 'green',
    defaultText: 'white',
    defaultDisabled: undefined,
    defaultHover: undefined,
    defaultSelected: undefined,
    primary: 'red',
    primaryText: 'white',
    primaryDisabled: undefined,
    primaryHover: undefined,
    primarySelected: undefined,
    secondary: 'blue',
    secondaryText: 'white',
    secondaryDisabled: undefined,
    secondaryHover: undefined,
    secondarySelected: undefined,
    disabled: 'lightgray',
    shadow: 'black',
  },
};
