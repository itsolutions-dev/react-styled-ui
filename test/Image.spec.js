import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import 'styled-components-test-utils/lib/jest';
import theme from '../src/theme';
import Image from '../src/Image';

describe('Image', () => {
  test('should render an Image', () => {
    const component = ReactTestRenderer.create(<Image theme={theme} />);
    expect(component).toBeDefined();
  });

  test('should have an img tag', () => {
    const component = ReactTestRenderer.create(<Image theme={theme} />).toJSON();
    expect(component.type).toEqual('img');
  });
});
