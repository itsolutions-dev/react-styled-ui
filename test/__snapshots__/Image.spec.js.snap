// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render an Image 1`] = `
.c0 {
  display: block;
  max-width: 100%;
  height: auto;
}

<img
  className="c0"
/>
`;
